base/linalg/cholesky.jl

```diff
@@ -491,6 +491,21 @@ function A_ldiv_B!(C::CholeskyPivoted, B::StridedMatrix)
     end
 end
 
+\(::Cholesky, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::CholeskyPivoted, ::RowVector) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
+\(::Cholesky{<:Number}, ::RowVector{<:Number}) = throw(DimensionMismatch("Cannot left-divide matrix by transposed vector"))
```

Comment:
So it's been decided (for now at least) that we allow `vector*matrix` and `matrix*rowvector` if matrix is `n × 1` or `1 × n`, respectively (my original PR for rowvector forbade both of these, as well as the associated `\` and `/` methods). I suppose we should be consistent between `*` and `\` and `/` and remove these for cholesky and bidiagonal and everywhere else too. Unfortunately these here are a bit useless - they can only possibly work for `1×1` cholesky and length-1 rowvector (I even recently found out base supports 1-vector * 1-vector -> 1×n matrix!)

@stevengj - thoughts?